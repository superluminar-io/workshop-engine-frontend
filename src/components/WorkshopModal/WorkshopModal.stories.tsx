import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { WorkshopFormData, WorkshopModal } from "./WorkshopModal";
import { useForm } from "react-hook-form";

export default {
  title: "WorkshopModal",
  component: WorkshopModal,
  argTypes: {
    onSubmit: {
      action: "submitted",
    },
    onCancel: {
      action: "cancelled",
    },
    control: {
      table: {
        disable: true,
      },
    },
  },
} as ComponentMeta<typeof WorkshopModal>;

const Template: ComponentStory<typeof WorkshopModal> = (args) => {
  const methods = useForm<WorkshopFormData>();

  return <WorkshopModal control={methods.control} {...args} />;
};

export const Main = Template.bind({});
Main.storyName = "default";
Main.args = {
  open: true,
  title: "Create new workshop",
  submitButtonLabel: "Create",
};

export const Loading = Template.bind({});
Loading.storyName = "loading";
Loading.args = {
  open: true,
  loading: true,
  title: "Create new workshop",
  submitButtonLabel: "Create",
};
