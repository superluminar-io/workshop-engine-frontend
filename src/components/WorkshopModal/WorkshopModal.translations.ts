export const inputTitleLabel = "Title";
export const inputTitlePlaceholder = "Serverless Workshop";

export const inputDescriptionLabel = "Description (Markdown supported)";

export const inputAwsAccountIdLabel = "AWS Account ID (12-digit number)";
export const inputAwsAccountIdPlaceholder = "518170230839";

export const inputAttendeesLabel = "Attendees (comma separated)";
export const inputAttendeesPlaceholder = "abc@example.com, def@example.com";

export const cancelButtonLabel = "Cancel";
