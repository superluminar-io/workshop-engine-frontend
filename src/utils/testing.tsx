import React from "react";
import {
  render as renderTestingLibrary,
  RenderOptions,
  RenderResult,
} from "@testing-library/react";
import { ThemeProvider } from "@mui/material";
import { theme } from "../config/theme";

export const noop = () => null;

export const render = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => {
  return renderTestingLibrary(
    <ThemeProvider theme={theme}>{ui}</ThemeProvider>,
    options
  );
};
