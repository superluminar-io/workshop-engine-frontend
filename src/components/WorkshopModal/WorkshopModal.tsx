import React, { FormEvent, FormEventHandler } from "react";
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  useMediaQuery,
  useTheme,
} from "@mui/material";
import LoadingButton from "@mui/lab/LoadingButton";
import { Control } from "react-hook-form";
import { ControlledTextField } from "../ControlledTextField/ControlledTextField";
import * as translations from "./WorkshopModal.translations";

export interface WorkshopFormData {
  title: string;
  awsAccountId?: string;
  // comma separated
  attendees?: string;
}

export interface WorkshopModalProps {
  title: string;
  submitButtonLabel: string;
  open: boolean;
  loading: boolean;
  control: Control<WorkshopFormData>;
  onCancel(): void;
  onSubmit: FormEventHandler<HTMLFormElement>;
}

export const WorkshopModal: React.FunctionComponent<WorkshopModalProps> = ({
  title,
  submitButtonLabel,
  open,
  loading,
  onCancel,
  control,
  onSubmit,
}) => {
  const theme = useTheme();
  const fullScreen = useMediaQuery(theme.breakpoints.down("md"));

  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    onSubmit(event);
  };

  return (
    <Dialog
      open={open}
      onClose={loading ? undefined : onCancel}
      fullWidth
      maxWidth="sm"
      fullScreen={fullScreen}
    >
      <form onSubmit={handleSubmit}>
        <DialogTitle>{title}</DialogTitle>
        <DialogContent>
          <ControlledTextField
            control={control}
            type="text"
            name="title"
            rules={{
              required: true,
            }}
            autoFocus
            margin="dense"
            label={translations.inputTitleLabel}
            fullWidth
            required={true}
            placeholder={translations.inputTitlePlaceholder}
            disabled={loading}
          />
          <ControlledTextField
            control={control}
            type="text"
            margin="dense"
            name="awsAccountId"
            label={translations.inputAwsAccountIdLabel}
            placeholder={translations.inputAwsAccountIdPlaceholder}
            inputProps={{ pattern: "[0-9]{12}" }}
            fullWidth
            disabled={loading}
          />
          <ControlledTextField
            control={control}
            margin="dense"
            name="attendees"
            label={translations.inputAttendeesLabel}
            type="email"
            fullWidth
            placeholder={translations.inputAttendeesPlaceholder}
            inputProps={{ multiple: true }}
            disabled={loading}
          />
        </DialogContent>
        <DialogActions sx={{ pr: 3, pl: 3, pb: 2 }}>
          <Button
            variant="outlined"
            onClick={onCancel}
            fullWidth={fullScreen}
            disabled={loading}
          >
            {translations.cancelButtonLabel}
          </Button>
          <LoadingButton
            variant="contained"
            type="submit"
            fullWidth={fullScreen}
            loading={loading}
          >
            {submitButtonLabel}
          </LoadingButton>
        </DialogActions>
      </form>
    </Dialog>
  );
};
