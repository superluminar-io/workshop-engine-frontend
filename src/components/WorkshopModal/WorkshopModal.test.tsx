import { fireEvent, screen } from "@testing-library/react";
import { useForm } from "react-hook-form";
import { noop, render } from "../../utils/testing";
import { WorkshopFormData, WorkshopModal } from "./WorkshopModal";
import * as translations from "./WorkshopModal.translations";

const title = "Create new workshop";
const submitButtonLabel = "Create";

const TestComponent: React.FunctionComponent<{
  open: boolean;
  loading: boolean;
  onSubmit?(): void;
  onCancel?(): void;
}> = ({ open, loading, onSubmit, onCancel }) => {
  const methods = useForm<WorkshopFormData>();
  return (
    <WorkshopModal
      title={title}
      submitButtonLabel={submitButtonLabel}
      control={methods.control}
      open={open}
      loading={loading}
      onSubmit={onSubmit || noop}
      onCancel={onCancel || noop}
    />
  );
};

describe("modal open", () => {
  it("renders the modal", () => {
    render(<TestComponent open={true} loading={false} />);
    expect(document.body).toMatchSnapshot();
  });
});

describe("modal close", () => {
  it("renders the modal", () => {
    render(<TestComponent open={false} loading={false} />);
    expect(document.body).toMatchSnapshot();
  });
});

describe("loading", () => {
  it("renders the modal", () => {
    render(<TestComponent open={false} loading={true} />);
    expect(document.body).toMatchSnapshot();
  });
});

describe("click on submit button", () => {
  it("triggers the submit handler", async () => {
    const submitHandler = jest.fn();
    render(
      <TestComponent open={true} loading={false} onSubmit={submitHandler} />
    );

    // Set workshop title for a valid form submission
    fireEvent.change(
      screen.getByPlaceholderText(translations.inputTitlePlaceholder),
      {
        target: { value: "Serverless Workshop" },
      }
    );

    fireEvent.click(screen.getByText(submitButtonLabel));

    expect(submitHandler).toHaveBeenCalled();
  });
});

describe("click on cancel button", () => {
  it("triggers the canel handler", async () => {
    const cancelHandler = jest.fn();
    render(
      <TestComponent open={true} loading={false} onCancel={cancelHandler} />
    );

    fireEvent.click(screen.getByText(translations.cancelButtonLabel));

    expect(cancelHandler).toHaveBeenCalled();
  });
});
