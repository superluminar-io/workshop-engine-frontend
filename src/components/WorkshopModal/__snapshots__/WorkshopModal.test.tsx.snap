// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading renders the modal 1`] = `
<body
  style=""
>
  <div />
</body>
`;

exports[`modal close renders the modal 1`] = `
<body
  style=""
>
  <div />
</body>
`;

exports[`modal open renders the modal 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
}

@media print {
  .emotion-0 {
    position: absolute!important;
  }
}

.emotion-1 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.5);
  -webkit-tap-highlight-color: transparent;
  z-index: -1;
}

.emotion-2 {
  height: 100%;
  outline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media print {
  .emotion-2 {
    height: auto;
  }
}

.emotion-3 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 12px;
  box-shadow: none;
  margin: 32px;
  position: relative;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: calc(100% - 64px);
  max-width: 600px;
  width: calc(100% - 64px);
}

@media print {
  .emotion-3 {
    overflow-y: visible;
    box-shadow: none;
  }
}

@media (max-width:663.95px) {
  .emotion-3.MuiDialog-paperScrollBody {
    max-width: calc(100% - 64px);
  }
}

.emotion-4 {
  margin: 0;
  font-family: "Source Sans Pro",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  padding: 16px 24px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.emotion-5 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-overflow-scrolling: touch;
  overflow-y: auto;
  padding: 20px 24px;
}

.MuiDialogTitle-root+.emotion-5 {
  padding-top: 0;
}

.emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  margin-top: 8px;
  margin-bottom: 4px;
  width: 100%;
}

.emotion-7 {
  color: rgba(0, 0, 0, 0.6);
  font-family: "Source Sans Pro",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: calc(133% - 24px);
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: translate(14px, -9px) scale(0.75);
  -moz-transform: translate(14px, -9px) scale(0.75);
  -ms-transform: translate(14px, -9px) scale(0.75);
  transform: translate(14px, -9px) scale(0.75);
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  z-index: 1;
  pointer-events: auto;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-7.Mui-focused {
  color: #ff3b61;
}

.emotion-7.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-7.Mui-error {
  color: #d32f2f;
}

.emotion-8.Mui-error {
  color: #d32f2f;
}

.emotion-9 {
  font-family: "Source Sans Pro",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  position: relative;
  border-radius: 12px;
}

.emotion-9.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-9:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.87);
}

@media (hover: none) {
  .emotion-9:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-9.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #ff3b61;
  border-width: 2px;
}

.emotion-9.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #d32f2f;
}

.emotion-9.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.26);
}

.emotion-10 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding: 16.5px 14px;
}

.emotion-10::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-10::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-10:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-10::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-10:focus {
  outline: 0;
}

.emotion-10:invalid {
  box-shadow: none;
}

.emotion-10::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-10:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-10.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-10:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-10:-webkit-autofill {
  border-radius: inherit;
}

.emotion-11 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-12 {
  float: unset;
  overflow: hidden;
  display: block;
  width: auto;
  padding: 0;
  height: 11px;
  font-size: 0.75em;
  visibility: hidden;
  max-width: 100%;
  -webkit-transition: max-width 100ms cubic-bezier(0.0, 0, 0.2, 1) 50ms;
  transition: max-width 100ms cubic-bezier(0.0, 0, 0.2, 1) 50ms;
  white-space: nowrap;
}

.emotion-12>span {
  padding-left: 5px;
  padding-right: 5px;
  display: inline-block;
  opacity: 0;
  visibility: visible;
}

.emotion-14 {
  color: rgba(0, 0, 0, 0.6);
  font-family: "Source Sans Pro",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: calc(100% - 24px);
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: translate(14px, 16px) scale(1);
  -moz-transform: translate(14px, 16px) scale(1);
  -ms-transform: translate(14px, 16px) scale(1);
  transform: translate(14px, 16px) scale(1);
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  z-index: 1;
  pointer-events: none;
}

.emotion-14.Mui-focused {
  color: #ff3b61;
}

.emotion-14.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-14.Mui-error {
  color: #d32f2f;
}

.emotion-18 {
  float: unset;
  overflow: hidden;
  display: block;
  width: auto;
  padding: 0;
  height: 11px;
  font-size: 0.75em;
  visibility: hidden;
  max-width: 0.01px;
  -webkit-transition: max-width 50ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: max-width 50ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  white-space: nowrap;
}

.emotion-18>span {
  padding-left: 5px;
  padding-right: 5px;
  display: inline-block;
  opacity: 0;
  visibility: visible;
}

.emotion-25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-right: 24px;
  padding-left: 24px;
  padding-bottom: 16px;
}

.emotion-25>:not(:first-of-type) {
  margin-left: 8px;
}

.emotion-26 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Source Sans Pro",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 5px 15px;
  border-radius: 12px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border: 1px solid rgba(255, 59, 97, 0.5);
  color: #ff3b61;
}

.emotion-26::-moz-focus-inner {
  border-style: none;
}

.emotion-26.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-26 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-26:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(255, 59, 97, 0.04);
  border: 1px solid #ff3b61;
}

@media (hover: none) {
  .emotion-26:hover {
    background-color: transparent;
  }
}

.emotion-26.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

.emotion-27 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Source Sans Pro",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 12px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #ff3b61;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 1px 2px 0px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-27::-moz-focus-inner {
  border-style: none;
}

.emotion-27.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-27 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-27:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(178, 41, 67);
  box-shadow: rgba(0, 0, 0, 0.1) 0px 4px 6px -1px,rgba(0, 0, 0, 0.06) 0px 2px 4px -1px;
}

@media (hover: none) {
  .emotion-27:hover {
    background-color: #ff3b61;
  }
}

.emotion-27:active {
  box-shadow: none;
}

.emotion-27.Mui-focusVisible {
  box-shadow: rgba(0, 0, 0, 0.1) 0px 20px 25px -5px,rgba(0, 0, 0, 0.04) 0px 10px 10px -5px;
}

.emotion-27.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-27 .MuiLoadingButton-startIconLoadingStart,
.emotion-27 .MuiLoadingButton-endIconLoadingEnd {
  -webkit-transition: opacity 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  opacity: 0;
}

.emotion-27.MuiLoadingButton-loading {
  color: transparent;
}

<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiModal-root MuiDialog-root emotion-0"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root emotion-1"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper emotion-2"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby=":r0:"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth emotion-3"
        role="dialog"
      >
        <form>
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root emotion-4"
            id=":r0:"
          >
            Create new workshop
          </h2>
          <div
            class="MuiDialogContent-root emotion-5"
          >
            <div
              class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root emotion-6"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused Mui-required emotion-7"
                data-shrink="true"
                for=":r1:"
                id=":r1:-label"
              >
                Title
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk emotion-8"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl emotion-9"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input emotion-10"
                  id=":r1:"
                  name="title"
                  placeholder="Serverless Workshop"
                  required=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline emotion-11"
                >
                  <legend
                    class="emotion-12"
                  >
                    <span>
                      Title
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root emotion-6"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary emotion-14"
                data-shrink="false"
                for=":r2:"
                id=":r2:-label"
              >
                AWS Account ID (12-digit number)
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl emotion-9"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input emotion-10"
                  id=":r2:"
                  name="awsAccountId"
                  pattern="[0-9]{12}"
                  placeholder="518170230839"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline emotion-11"
                >
                  <legend
                    class="emotion-18"
                  >
                    <span>
                      AWS Account ID (12-digit number)
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root emotion-6"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary emotion-14"
                data-shrink="false"
                for=":r3:"
                id=":r3:-label"
              >
                Attendees (comma separated)
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl emotion-9"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input emotion-10"
                  id=":r3:"
                  multiple=""
                  name="attendees"
                  placeholder="abc@example.com, def@example.com"
                  type="email"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline emotion-11"
                >
                  <legend
                    class="emotion-18"
                  >
                    <span>
                      Attendees (comma separated)
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing emotion-25"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  emotion-26"
              tabindex="0"
              type="button"
            >
              Cancel
            </button>
            <button
              class="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root emotion-27"
              id=":r4:"
              tabindex="0"
              type="submit"
            >
              Create
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
